libbpf-rs/src/map.rs

```diff
@@ -336,3 +342,37 @@ pub enum MapType {
     /// using this library is a bit out of date.
     Unknown = u32::MAX,
 }
+
+pub struct MapKeyIter<'a> {
+    map: &'a Map,
+    prev: Option<Vec<u8>>,
+    next: Vec<u8>,
+}
+
+impl<'a> MapKeyIter<'a> {
+    fn new(map: &'a Map, key_size: u32) -> Self {
+        Self {
+            map,
+            prev: None,
+            next: vec![0; key_size as usize],
+        }
+    }
+}
+
+impl<'a> Iterator for MapKeyIter<'a> {
+    type Item = Vec<u8>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let prev = self.prev.as_ref().map_or(ptr::null(), |p| p.as_ptr());
+
+        if unsafe {
+            libbpf_sys::bpf_map_get_next_key(self.map.fd(), prev as _, self.next.as_mut_ptr() as _)
+        } != 0
+        {
```

Comment:
nit: this is quite a convoluted formatting, why not have a local variable storing the returned error?